proxy.php

```diff
@@ -159,41 +175,71 @@ protected function isCached($request)
 	 *
 	 * @access protected
 	 * @param string $request The image to cache/validate
-	 * @return bool|int Whether the specified image was cached or error code when accessing
+	 * @return int -1 error, 0 too big, 1 valid image
 	 */
 	protected function cacheImage($request)
 	{
 		$dest = $this->getCachedPath($request);
-
 		$curl = new curl_fetch_web_data(array(CURLOPT_BINARYTRANSFER => 1));
-		$request = $curl->get_url_data($request);
-		$responseCode = $request->result('code');
-		$response = $request->result();
-
-		if (empty($response))
-			return false;
+		$curl_request = $curl->get_url_data($request);
+		$responseCode = $curl_request->result('code');
+		$response = $curl_request->result();
```

Comment:
Right now, the proxy only works if the curl library was included when PHP was compiled (it usually is, of course, but not always). It would be nice to include here [some of the fallback logic](https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Subs-Package.php#L3141-L3253) from the `fetch_web_data()` function in Subs-Package.php.

Reply:
Well the whole subs* stuff is my eyes is to mutch overhead for this that we don't needs this in the common case,
on the second hand we provide with the external proxy stuff a second solutions.
When your enverment doesn't met this requirement than it's "bad luck".

Reply:
But if you would prefer not to do this, that's fine. I can do it myself afterwards. :)

Reply:
Feel free to add this after this pr.

Reply:
Okay, that's fine.